src/shared/internal-calendar/internal-week-selection-calendar-day.style.tsx

```diff
@@ -0,0 +1,340 @@
+        }
+    }}
+
+    ${(props) => {
+        const { $hovered, $selected } = props;
+
+        if ($hovered) {
+            return css`
+                border-top: 1px dashed ${Color.Accent.Light[4]};
+                border-bottom: 1px dashed ${Color.Accent.Light[4]};
+                background-color: ${Color.Accent.Light[6]};
+            `;
+        }
+
+        if ($selected) {
+            return css`
+                border-top: 1px solid ${Color.Primary};
+                border-bottom: 1px solid ${Color.Primary};
+                background-color: ${Color.Accent.Light[5]};
+            `;
+        }
+    }}
+
+
+    ${(props) => {
+        if (props.$overlap) {
+            return css`
+                border-top: 1px solid ${Color.Primary};
+                border-bottom: 1px solid ${Color.Primary};
+                background-color: ${Color.Neutral[8]};
+                box-shadow: 4px 0px 4px 0px ${Color.Shadow.Accent};
+            `;
+        }
+    }}
+`;
+
+export const OverflowCircle = styled.div<OverflowCircleProps>`
+    display: none;
+    position: absolute;
+    pointer-events: none;
+    background-color: transparent;
+    width: 2.5rem;
+    height: 2.5rem;
+    border-radius: 50%;
+
+    ::before,
+    ::after {
+        content: "";
+        display: block;
+        position: absolute;
+        width: calc(3.5rem / 2);
+        pointer-events: none;
+        height: 100%;
+    }
+
+    ${(props) => {
+        if (props.$hovered) {
+            return css`
+                border: 1px dashed ${Color.Accent.Light[4]};
+
+                ::before {
+                    background-color: ${Color.Accent.Light[6]};
+                    top: 0;
+                }
+            `;
+        }
+    }}
+
+    ${(props) => {
+        if (props.$selected) {
+            return css`
+                border: 1px solid ${Color.Primary};
+
+                ::before {
+                    background-color: ${Color.Accent.Light[5]};
+                    top: 0;
+                }
+            `;
+        }
+    }}
+
+    ${(props) => {
+        if (props.$overlap) {
+            return css`
+                ::before {
+                    background-color: ${Color.Neutral[8]};
+                    top: 0;
+                }
+            `;
+        }
+    }}
+
+    ${(props) => {
+        if (props.$overlap && props.$position === "left") {
+            return css`
+                box-shadow: -1px 0px 4px 0px ${Color.Shadow.Accent};
+            `;
+        } else if (props.$overlap && props.$position === "right") {
+            return css`
+                box-shadow: 1px 0px 4px 0px ${Color.Shadow.Accent};
+            `;
+        }
+    }}
+
+    ${(props) => {
+        switch (props.$position) {
+            case "left":
+                return css`
+                    display: block;
+                    border-right-color: transparent;
+
+                    ::before {
+                        right: -0.45rem;
```

Comment:
you can do `left: 50%;` here if you want to anchor the element from the middle to the right of the circle and `right: 50%` below

Reply:
sadly part was the border circle shadow is large than the square.

so it related the `3.5rem` above also. Added bigger width to hide the outstanding shadow.